### DISCUSSION TITLE
How to do static hosting in v3?

### DISCUSSION THREAD

**Discussion by Erudition** [dillonkearns/elm-pages#452] [Q&A]
I recently upgraded to v3, and while the site works fine in dev mode, the build just contains `index.html` and none of my other routes. So, the homepage appears online, but every link is a 404.

Inspecting the build, it looks like there's a bunch of files relating to redirects that are supposed to be followed, and most of these are built for Netlify. I'm not using Netlify, nor any sort of "serverless" setup, just a static site hosted on GitLab Pages.

How do I get `elm-pages build` to build a normal static site like in v2? The docs don't say much, but they do suggest that static site building is still supported.

**Comment by Erudition** [comment]
Okay so I realized there's more docs than the website. The website explains a lot of high-level stuff but doesn't tell me, for example, that I can't use the script for adding a new page if I want it to be a "prerendered route". I managed to copy the Index.elm code to get a second page working...

I can't view the local docs because of #445 , so I had to inspect the code of the hidden RouteBuilder.elm file :grimacing:

**Reply by dillonkearns** [reply]
Happy to accept a PR if there are any opportunities for making that more clear in the docs that you can think of.

**Comment by dillonkearns** [comment]
There is an [`AddStaticRoute`](https://github.com/dillonkearns/elm-pages-starter/blob/29143e11d2d77663e80b06c87f93b77b40bce518/script/src/AddStaticRoute.elm) script in the starter repo as of a few weeks ago. It makes it a lot easier to generate static routes.
